python/packages/autogen-core/samples/adas/README.md

```diff
@@ -0,0 +1,515 @@
+
+### Meta-Agent will try again fresh if it encounters (code compilation) errors when trying to execute
+
+An example of an exception is the following:
+```
+Error during evaluation:\nClosureAgent.register() takes 4 positional arguments but 5 positional arguments (and 1 keyword-only argument) were given\nCarefully consider where you went wrong in your latest implementation. Using insights from previous attempts, try to debug the current code to implement the same thought. Repeat your previous thought in 'thought', and put your thinking for debugging in 'debug_thought'", source='adas_agent')
+```
+The current behavior is that the meta-agent will start over the generation sequence (using the 5 steps of LLM calls).
+
+Note: The `adas.py` script can still get stuck even if the code of the agent system compiles. This can occur because the agent system itself hangs due to a message being published to a topic that is not used. See this [section](#the-code-that-the-meta-agent-does-not-compile) under the Troubleshooting section for details on how to address this issue.
+
+### Notable arguments to the script
+
+Please see the `adas.py` file for details of all available settings.
+
+- `data_filename`: the name of full path of the dataset location
+- `benchmark_specific_utils_file`: Benchmark-specific utility file to load the dataset and also evaluate the outputs. This file must contain the `load_dataset` and `compute_metrics` functions
+- `meta_agent_model_config`: JSON string of the `AzureOpenAIChatCompletionClient` settings for the Meta-Agent.
+- `base_agent_model_config`: JSON string of the `AzureOpenAIChatCompletionClient` settings for the Base Agent.
+- `n_generation`: number of generations of new agents that the meta-agent tries to discover
+- `expr_name`: name of the output file containing both the original/seed and newly generated agent systems, as well as their fitness scores.
+- `max_workers`: the number of threads to spin up in a `ThreadPoolExecutor`, to parallelize the execution of the particular Agent System that is currently being evaluated.
+
+## QuickStart
+
+### Install AutoGen-Core API
+
+Follow the instructions here: [Installation — AutoGen](https://github.com/microsoft/autogen.git). Here is a summary:
+
+```bash
+python3 -m venv .venv
+source .venv/bin/activate
+
+# Clone the ADAS repo to be able to use the DROP dataset for the sample.
+# Recommended to perform a demo, though not required if you do not plan on evaluating with DROP, and intend to run with your own dataset / benchmark.
+git clone https://github.com/ShengranHu/ADAS.git && cd ..
+
+# Clone the AutoGen package, and switch to branch with the adas script.
+git clone -b yeandy_adas https://github.com/yeandy/autogen.git
+cd autogen/python
+
+# Install autogen-core and autogen-ext in editable mode
+pip install -e packages/autogen-core
+pip install -e packages/autogen-ext
+pip install -r packages/autogen-core/samples/adas/requirements.txt
```

Comment:
id also include pandas and numpy as you are using them in the example code

Reply:
I have `numpy` in the `requirements.txt`. Did you find yourself to have to separately install `pandas`?